src/ripple/conditions/impl/Der.h

```diff
@@ -0,0 +1,2798 @@
+          0x02, 0x82, 0x42, 0xC2, 0x22, 0xA2, 0x62, 0xE2, 0x12, 0x92, 0x52, 0xD2, 0x32, 0xB2, 0x72, 0xF2, 
+          0x0A, 0x8A, 0x4A, 0xCA, 0x2A, 0xAA, 0x6A, 0xEA, 0x1A, 0x9A, 0x5A, 0xDA, 0x3A, 0xBA, 0x7A, 0xFA,
+          0x06, 0x86, 0x46, 0xC6, 0x26, 0xA6, 0x66, 0xE6, 0x16, 0x96, 0x56, 0xD6, 0x36, 0xB6, 0x76, 0xF6, 
+          0x0E, 0x8E, 0x4E, 0xCE, 0x2E, 0xAE, 0x6E, 0xEE, 0x1E, 0x9E, 0x5E, 0xDE, 0x3E, 0xBE, 0x7E, 0xFE,
+          0x01, 0x81, 0x41, 0xC1, 0x21, 0xA1, 0x61, 0xE1, 0x11, 0x91, 0x51, 0xD1, 0x31, 0xB1, 0x71, 0xF1,
+          0x09, 0x89, 0x49, 0xC9, 0x29, 0xA9, 0x69, 0xE9, 0x19, 0x99, 0x59, 0xD9, 0x39, 0xB9, 0x79, 0xF9, 
+          0x05, 0x85, 0x45, 0xC5, 0x25, 0xA5, 0x65, 0xE5, 0x15, 0x95, 0x55, 0xD5, 0x35, 0xB5, 0x75, 0xF5,
+          0x0D, 0x8D, 0x4D, 0xCD, 0x2D, 0xAD, 0x6D, 0xED, 0x1D, 0x9D, 0x5D, 0xDD, 0x3D, 0xBD, 0x7D, 0xFD,
+          0x03, 0x83, 0x43, 0xC3, 0x23, 0xA3, 0x63, 0xE3, 0x13, 0x93, 0x53, 0xD3, 0x33, 0xB3, 0x73, 0xF3, 
+          0x0B, 0x8B, 0x4B, 0xCB, 0x2B, 0xAB, 0x6B, 0xEB, 0x1B, 0x9B, 0x5B, 0xDB, 0x3B, 0xBB, 0x7B, 0xFB,
+          0x07, 0x87, 0x47, 0xC7, 0x27, 0xA7, 0x67, 0xE7, 0x17, 0x97, 0x57, 0xD7, 0x37, 0xB7, 0x77, 0xF7, 
+          0x0F, 0x8F, 0x4F, 0xCF, 0x2F, 0xAF, 0x6F, 0xEF, 0x1F, 0x9F, 0x5F, 0xDF, 0x3F, 0xBF, 0x7F, 0xFF
+        };
+        return lut[b];
+    }
+
+    /** return the number of leading zero bytes before the last byte
+
+        @note If no bits are set on a 64-bit integer, this function returns 7
+        _not_ 8 because der will always consider the last byte, even if it is
+        zero.
+     */
+    static
+    std::size_t
+    numLeadingZeroBytes(std::bitset<S> const& s)
+    {
+        std::size_t result = 0;
+        auto curByteIndex = maxBytes;
+        auto const bits = s.to_ulong();
+        if (bits == 1)
+            // Always consider the last byte, even if it is zero
+            return maxBytes - 1;
+        while (curByteIndex--)
+        {
+            std::uint8_t const b = (bits >> curByteIndex * 8) & 0xff;
+            if (b)
+                return result;
+            ++result;
+        }
+        return result-1;
+    }
+
+    static
+    std::uint8_t
+    numUnusedBits(
+        std::bitset<S> const& s,
+        std::size_t leadingZeroBytes)
+    {
+        // b is first non-zero byte
+        auto const bits = s.to_ulong ();
+        std::uint8_t const b =
+            (bits >> (maxBytes - leadingZeroBytes - 1) * 8) & 0xff;
+        if (b & 0x80)
+            return 0;
+        if (b & 0x40)
+            return 1;
+        if (b & 0x20)
+            return 2;
+        if (b & 0x10)
+            return 3;
+        if (b & 0x08)
+            return 4;
+        if (b & 0x04)
+            return 5;
+        if (b & 0x02)
+            return 6;
+        if (b & 0x01)
+            return 7;
+        // der always considers the last bit, even if no bits are set
+        return 7;
```

Comment:
Agreed it's not necessary, but I'd like to keep it anyway. The code reads better to me as "if this bit is set, return this value". Even though the last bit is somewhat special, I'd like to keep the same pattern as the other bit checks.